pxr/usdImaging/usdImaging/collectionCache.cpp

```diff
@@ -78,15 +97,16 @@ UsdImaging_CollectionCache::UpdateCollection(UsdCollectionAPI const& c)
     _pathsForQuery[query].insert(path);
     _idForPath[path] = id;
 
+    _MarkCollectionContentDirty(stage, query);
+    
```

Comment:
I feel like there's some missing bits here...  If _queryForId[id] before you update it != query, also _MarkContentDirty(stage, _queryForId[id])?

Reply:
The `RemoveCollection` call at the beginning of `UpdateCollection` will mark the previous collection content dirty, wouldn't this cover it?